app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -34,8 +34,9 @@ def save_inventory_collection!(inventory_collection, association)
         inventory_objects_index[index] = inventory_object
       end
 
-      unique_index_keys = inventory_collection.manager_ref_to_cols
-      unique_db_indexes = Set.new
+      unique_index_keys      = inventory_collection.manager_ref_to_cols
+      unique_db_indexes      = Set.new
+      unique_db_primary_keys = Set.new
```

Comment:
primary_key should be always :id, you can have any other combination of columns that will make a unique_key. So primary key constraint and unique index constraint are 2 different things. :-) That wiki page is somewhat wrong :-)